src/java/org/apache/cassandra/auth/MutualTlsWithPasswordFallbackAuthenticator.java

```diff
@@ -46,13 +46,21 @@ public void setup()
         mutualTlsAuthenticator.setup();
     }
 
+    @Override
+    public boolean supportsEarlyCertificateAuthentication()
```

Comment:
We set `supportsEarlyCertificateAuthentication` to true here because if the client provides a certificate, Cassandra will try to verify against it.  If not, it will send an `AUTHENTICATE` message back tot he client, and continue with the usual password-based auth flow.